OutOfSchool/OutOfSchool.WebApi/Services/AreaAdminService.cs

```diff
@@ -103,7 +104,14 @@ public async Task<AreaAdminDto> GetByUserId(string id)
             Logger.LogDebug(
                 "AreaAdmin creating is not possible. Username {Email} is already taken",
                 areaAdminBaseDto.Email);
-            throw new InvalidOperationException($"Username {areaAdminBaseDto.Email} is already taken.");
+            return new ErrorResponse
+            {
+                HttpStatusCode = HttpStatusCode.BadRequest,
+                ApiErrorResponse = new ApiErrorResponse(new List<ApiError>()
+                {
+                    ApiErrorsTypes.Common.EmailAlreadyTaken("AreaAdmin", areaAdminBaseDto.Email),
+                }),
```

Comment:
(Code duplication, suggestion) It looks like we have a repetitive pattern:
```
    ... new ApiErrorResponse(new List<ApiError>()
    {
        .../*construct single specific error*/... ,
    })
```
What if extract it to some extension method (i.e. `ToResponse()`) that could be used like
```
  return new ErrorResponse
  {
      HttpStatusCode = HttpStatusCode.BadRequest,
      ApiErrorResponse = ApiErrorsTypes.Common.EmailAlreadyTaken("AreaAdmin", areaAdminBaseDto.Email).ToResponse()
  }
```

Additionally another pattern is pretty common
```
    return new ErrorResponse
    {
        HttpStatusCode = HttpStatusCode.BadRequest,
        ApiErrorResponse = .../*some ApiErrorResponse*/...,
    };
```
Couldn't it be extracted as some "builder" method of `ErrorResponse`, i.e.
```
    return ErrorResponse.BadRequest( .../*some ApiErrorResponse*/...);
```

Reply:
(Please don't resolve conversations I've started. I do it after my review. If you need to mark that you've done something you could either "react" on my message or write the comment below)

Ok, thanks you for implementation of second idea from previous comment (`ErrorResponse.BadRequest()`), but what about first one? 
Hav you disliked it or skipped it or ...? 

Asking this, because I see same pattern in your tests `SetUp()`'s

Reply:
I removed this pattern from SetUp()'s.

As for the first pattern, I changed it to look like this:

-"Creating list apiErrors"
-"Filling it with errors as needed"
-"Return errors, if any"

I made this decision because in the future the errors will not only apply to entering already existing mail when creating. There will still be such errors for phone numbers, and maybe something else.

I understood the solution with ToResponse() like this:
There is a scoped service to which we add errors during operation using the ToResponse() method. Such solutions have already been implemented by me, but you said that it was not ok, due to heavy testing.

Btw, sorry for resolving conversations.

Reply:
@VadymLevkovskyi

Reply:
> Btw, sorry for resolving conversations.
Np, it was just attempt to align communication.

Thank you for your changes, and your explanation.